frontend/apps/reader/modules/readerzooming.lua

```diff
@@ -6,18 +6,46 @@ local Geom = require("ui/geometry")
 local GestureRange = require("ui/gesturerange")
 local InfoMessage = require("ui/widget/infomessage")
 local InputContainer = require("ui/widget/container/inputcontainer")
+local SpinWidget = require("ui/widget/spinwidget")
 local UIManager = require("ui/uimanager")
 local logger = require("logger")
+local util = require("util")
 local _ = require("gettext")
 local Input = Device.input
 local Screen = Device.screen
 local T = require("ffi/util").template
 
 local ReaderZooming = InputContainer:new{
     zoom = 1.0,
+    available_zoom_modes = {
+        "page",
+        "pagewidth",
+        "pageheight",
+        "content",
+        "contentwidth",
+        "contentheight",
+        "columns",
+        "rows",
+        "manual",
+    },
     -- default to nil so we can trigger ZoomModeUpdate events on start up
     zoom_mode = nil,
     DEFAULT_ZOOM_MODE = "pagewidth",
+    -- for pan mode: fit to width/zoom_factor,
+    -- with overlap of zoom_overlap_h % (horizontally)
+    -- and zoom_overlap_v % (vertically).
+    zoom_factor = 2,
+    zoom_pan_settings = {
+        "zoom_factor",
+        "zoom_overlap_h",
+        "zoom_overlap_v",
+        "zoom_bottom_to_top",
+        "zoom_direction_vertical",
+    },
+    zoom_overlap_h = 40,
+    zoom_overlap_v = 40,
```

Comment:
This is scaleBySize-d elsewhere?

Reply:
This isn't in pixels, but in percent.

Reply:
Same difference, as long as it's behaving properly across DPI. ;-)